src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/InMemoryPartitionedUserTokenCacheAccessor.cs

```diff
@@ -0,0 +1,259 @@
+                    $"Cannot delete account because it was not found in the cache. Key {item.GetKey()}.",
+                    "Cannot delete account because it was not found in the cache");
+            }
+        }
+
+        #endregion
+
+        #region Get All
+        public IReadOnlyList<MsalAccessTokenCacheItem> GetAllAccessTokens(string partitionKey = null)
+        {
+            if (string.IsNullOrEmpty(partitionKey))
+            {
+                return AccessTokenCacheDictionary.SelectMany(dict => dict.Value).Select(kv => kv.Value).ToList();
+            }
+            else
+            {
+                AccessTokenCacheDictionary.TryGetValue(partitionKey, out ConcurrentDictionary<string, MsalAccessTokenCacheItem> partition);
+                return partition?.Select(kv => kv.Value)?.ToList() ?? new List<MsalAccessTokenCacheItem>();
+            }
+        }
+
+        public IReadOnlyList<MsalRefreshTokenCacheItem> GetAllRefreshTokens(string partitionKey = null)
+        {
+            if (string.IsNullOrEmpty(partitionKey))
+            {
+                return RefreshTokenCacheDictionary.SelectMany(dict => dict.Value).Select(kv => kv.Value).ToList();
+            }
+            else
+            {
+                RefreshTokenCacheDictionary.TryGetValue(partitionKey, out ConcurrentDictionary<string, MsalRefreshTokenCacheItem> partition);
+                return partition?.Select(kv => kv.Value)?.ToList() ?? new List<MsalRefreshTokenCacheItem>();
+            }
+        }
+
+        public IReadOnlyList<MsalIdTokenCacheItem> GetAllIdTokens(string partitionKey = null)
+        {
+            if (string.IsNullOrEmpty(partitionKey))
+            {
+                return IdTokenCacheDictionary.SelectMany(dict => dict.Value).Select(kv => kv.Value).ToList();
+            }
+            else
+            {
+                IdTokenCacheDictionary.TryGetValue(partitionKey, out ConcurrentDictionary<string, MsalIdTokenCacheItem> partition);
+                return partition?.Select(kv => kv.Value)?.ToList() ?? new List<MsalIdTokenCacheItem>();
+            }
+        }
+
+        public IReadOnlyList<MsalAccountCacheItem> GetAllAccounts(string partitionKey = null)
+        {
+            if (string.IsNullOrEmpty(partitionKey))
+            {
+                return AccountCacheDictionary.SelectMany(dict => dict.Value).Select(kv => kv.Value).ToList();
+            }
+            else
+            {
+                AccountCacheDictionary.TryGetValue(partitionKey, out ConcurrentDictionary<string, MsalAccountCacheItem> partition);
+                return partition?.Select(kv => kv.Value)?.ToList() ?? new List<MsalAccountCacheItem>();
+            }
+        }
+
+        public IReadOnlyList<MsalAppMetadataCacheItem> GetAllAppMetadata()
+        {
+            return AppMetadataDictionary.Select(kv => kv.Value).ToList();
+        }
+        #endregion
+
+        public void SetiOSKeychainSecurityGroup(string keychainSecurityGroup)
```

Comment:
can this be moved elsewhere? it is platform specific.

Reply:
Yes I agree, it should not be here. But this PR is already too large, and we should not attempt to fix everything.